Python scripting fails to load binary modules

When using a Python script in obs, loading a binary module fails due to non-exported CPython 
symbols.

I am using obs on a Linux Debian Buster.  Tested with obs compiled from current git master (ccb7d897) as well as the debian package provided in testing, aka  `obs-studio 24.0.3+dfsg1-2`.

Given the following example which tries to load the `termios` python package (a cython package):

```python
import termios
```
## Expected Behavior

I would expect this python snippet to load properly (and do nothing).

## Current Behavior

I fails with the following traceback:

```
Traceback (most recent call last):
File "/home/david/perso/elec/obs-tally/obsFirmataTally.py", line 9, in <module>
  import termios
ImportError: /usr/lib/python3.7/lib-dynload/termios.cpython-37m-x86_64-linux-gnu.so: undefined symbol: PyExc_TypeError
```

## Steps to Reproduce
1. create a `test_import.py` script with the content being just `import termios`
2. load the script in the Tools->Scripts dialog 
3. open the Script Log dialog to see the described  traceback

## Additional information 

Using Python 3.7 from the system (3.7.3-1).

It also fails to load numpy for example.

Ran across this thread while tracking down a funny issue with Debian Bullseye. I guess it's having trouble finding python. 

obs-scripting-python-import.h:39:10: fatal error: Python.h: No such file or directory
   39 | #include <Python.h>

**find / -iname python.h**
/usr/include/python3.9/Python.h
/usr/include/python2.7/Python.h
/usr/include/python3.8/Python.h

--- a/UI/CMakeLists.txt
+++ b/UI/CMakeLists.txt
@@ -148,6 +148,7 @@ elseif(APPLE)
 
 elseif(UNIX)
 	find_package(Qt5Gui REQUIRED)
+	find_package(PythonDeps QUIET)
 
 	include_directories(${Qt5Gui_PRIVATE_INCLUDE_DIRS})
 
@@ -158,6 +159,16 @@ elseif(UNIX)
 		${obs_PLATFORM_LIBRARIES}
 		Qt5::Gui
 		Qt5::GuiPrivate)
+
+	# python symbols must be in the global symbol table
+	# so we link the main executable to python if we expect
+	# obs-scripting python support to be enabled.
+	# see: https://github.com/obsproject/obs-studio/issues/2222 and https://bugs.python.org/issue36721
+	if(NOT DISABLE_PYTHON AND PYTHONLIBS_FOUND)
+		set(obs_PLATFORM_LIBRARIES
+			${obs_PLATFORM_LIBRARIES}
+			${PYTHON_LIBRARIES})
+	endif()
 
 	if("${CMAKE_SYSTEM_NAME}" MATCHES "FreeBSD")
 		list(APPEND obs_PLATFORM_LIBRARIES

